x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/common_model/ecs_event.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IEvent as IEventLogEvent } from '../../../../../../event_log/server';
+
+// https://www.elastic.co/guide/en/ecs/1.9/ecs-guidelines.html
+// https://www.elastic.co/guide/en/ecs/1.9/ecs-category-field-values-reference.html
+// https://www.elastic.co/guide/en/ecs/1.9/ecs-field-reference.html
+
+export type IEcsEvent = IEventLogEvent & IEcsAdditionalFields;
+
+interface IEcsAdditionalFields {
+  // https://www.elastic.co/guide/en/ecs/1.9/ecs-event.html
+  event?: {
```

Comment:
We already use a bunch of the `event` fields for fairly "generic" purposes, so I'm a little concerned about having application-specific data in here as well.  Not a problem now since there's no overlap, but I'm guessing there could be some day?

https://github.com/elastic/kibana/blob/6264c563d17bc72127adb200e06b3634a399463d/x-pack/plugins/event_log/generated/schemas.ts#L38-L48

Reply:
So, `IEcsAdditionalFields` just represents the fields we need to be supported in Event Log and which are currently missing in its schema. I mean, this PR is just a proposal, this interface is not going to be used in the working code as is and the missing fields will need to be moved to Event Log itself.

`event` fields here are all standard ECS fields ([spec](https://www.elastic.co/guide/en/ecs/1.9/ecs-event.html)), and the only custom field I propose is `kibana.detection_engine` :)

Not sure if I addressed your comment... Pls let me know :)